hack/update-deps.sh

```diff
@@ -20,6 +20,43 @@ set -o pipefail
 
 source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/library.sh
 
+# Remove two type of invalid symlinks
+# 1. Broken symlinks
+# 2. Symlinks pointing outside of the source tree
+# We use recursion to find the canonical path of a symlink
+# because both `realpath` and `readlink -m` are not available on Mac.
+function remove_invalid_symlink() {
```

Comment:
I think this is overkill. You don't need to traverse the directory tree, only check the target. As we discussed offline, parsing ls should be simple and do the trick.

Once this is merged, definitely deserves being moved to test-infra, together with a helper for the major features of update-deps.sh.

/hold

Reply:
Thanks a lot @adrcunha for the hint! 

On the other hand however, after using your changes there are some *valid* symlinks got deleted, see the following log:

``` bash
$ ./hack/update-deps.sh
unlinking broken:  ./vendor/github.com/knative/build/test/panic/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/test/workingdir/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/cmd/controller/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/cmd/logs/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/cmd/nop/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/cmd/git-init/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/cmd/creds-init/kodata/LICENSE
unlinking broken:  ./vendor/github.com/knative/build/cmd/webhook/kodata/LICENSE
unlinking broken:  ./vendor/github.com/prometheus/procfs/fixtures/self
```

However, e.g. https://github.com/knative/serving/blob/master/vendor/github.com/knative/build/test/panic/kodata/LICENSE is actually a valid symlink. 

Any reason why?

Reply:
There's a bug in my code, `-e ${target}` must be `-e ${link}`.